src/main/java/org/apache/commons/imaging/formats/gif/GifImageParser.java

```diff
@@ -407,7 +407,11 @@ public byte[] getIccProfileBytes(final ByteSource byteSource, final GifImagingPa
     @Override
     public ImageInfo getImageInfo(final ByteSource byteSource, final GifImagingParameters params)
             throws ImagingException, IOException {
-        final GifImageContents blocks = readFile(byteSource, false);
+        boolean stopReadingBeforeImageData = true;
```

Comment:
You've changed the default behavior here which makes the parameter a forced usage, which might be a surprised.